### PR TITLE
Migrate setup script into monorepo

### PR DESCRIPTION
**Pull request by lanej0**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Moves a setup script used when creating e2e test environments from a different repository to within the monorepo.

Closes # [WCQuality 926](https://github.com/woocommerce/woocommerce-quality/issues/926)

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Nothing to test

### CODE REVIEW

**Comment by adimoldovan** [comment]
Thanks for working on this @lanej0.
- We should also update the paths to the plugins to install to point to the ones in the monorepo instead of `https://github.com/rodelgc/woo-e2e-site-setup/raw/trunk/mu-plugins/...`.
- We should probably include woocommerce-cleanup.php as well.
- Ideally we'll have a single script, for both local and external environments setup. I don't see a good reason to maintain two scripts, they doing the same thing.

**Review by adimoldovan — CHANGES_REQUESTED** [review]
Thanks for working on this @lanej0.

- We should also update the paths to the plugins to install to point to the ones in the monorepo instead of https://github.com/rodelgc/woo-e2e-site-setup/raw/trunk/mu-plugins/....
- We should probably include woocommerce-cleanup.php as well.
- Ideally we'll have a single script, for both local and external environments setup. I don't see a good reason to maintain two scripts, they doing the same thing.

**Comment by lanej0** [comment]
Made the updates.

It was slightly more complicated because wp cli will only install a .zip file and the plugins in the monorepo were all just php files.  So I had to create zips and commit those to the repo so that we have somewhere to install from.

**Comment by adimoldovan** [comment]
> Made the updates.
> 
> It was slightly more complicated because wp cli will only install a .zip file and the plugins in the monorepo were all just php files. So I had to create zips and commit those to the repo so that we have somewhere to install from.

Instead of storing zip files in the repo we can download the plugins and zip them in the install script. The `zip` utility should be available and if not, it's a simple command to install.

```
curl -o filter-setter.php https://github.com/woocommerce/woocommerce/raw/trunk/plugins/woocommerce/tests/e2e-pw/bin/filter-setter.php
zip filter-setter.zip filter-setter.php
wp plugin install --force --activate filter-setter.zip
rm filter-setter.php filter-setter.zip
```

**Comment by lanej0** [comment]
> > Made the updates.
> > It was slightly more complicated because wp cli will only install a .zip file and the plugins in the monorepo were all just php files. So I had to create zips and commit those to the repo so that we have somewhere to install from.
> 
> Instead of storing zip files in the repo we can download the plugins and zip them in the install script. The `zip` utility should be available and if not, it's a simple command to install.
> 
> ```
> curl -o filter-setter.php https://github.com/woocommerce/woocommerce/raw/trunk/plugins/woocommerce/tests/e2e-pw/bin/filter-setter.php
> zip filter-setter.zip filter-setter.php
> wp plugin install --force --activate filter-setter.zip
> rm filter-setter.php filter-setter.zip
> ```

Yes!  I like this approach much better.  I was trying to envision some kind of GH action/workflow that would do the zip and then delete the files... but storage was stumping me.  Doing this part locally makes a lot of sense.

File: plugins/woocommerce/tests/e2e-pw/bin/test-env-setup-external.sh [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    # Download the PHP file
+    curl -o "$plugin.php" "https://raw.githubusercontent.com/woocommerce/woocommerce/trunk/plugins/woocommerce/tests/e2e-pw/bin/$plugin.php"
+
+    # Create a zip file
+    (chmod 755 "$plugin.php" && zip "${plugin%}.zip" "$plugin.php")
+
+    # Install the plugin from the local zip
+    wp plugin install --force --activate "${plugin%}.zip"
+
+    # rm "$plugin.php" "${plugin%}.zip"
```
Comment (adimoldovan) [inline plugins/woocommerce/tests/e2e-pw/bin/test-env-setup-external.sh]: Forgot it commented out?

File: plugins/woocommerce/tests/e2e-pw/bin/wp-cache-flush.php [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<?php
```
Comment (adimoldovan) [inline plugins/woocommerce/tests/e2e-pw/bin/wp-cache-flush.php]: I checked the setup guide that recommended adding the `wp_cache_flush();` snippet and it looks like the issue it was helping with was fixed in the meantime.
I don't think we need this and it should not be added.

File: plugins/woocommerce/tests/e2e-pw/bin/test-env-setup-external.sh (hunk 1 above)
Comment (lanej0) [reply plugins/woocommerce/tests/e2e-pw/bin/test-env-setup-external.sh]: Yup, was debugging.

**Review by adimoldovan — APPROVED** [review]

**Pull request merged**